src/patchkernel/surface_skd_tree.hpp

```diff
@@ -68,6 +72,15 @@ class SurfaceSkdTree : public PatchSkdTree {
     mutable std::vector<std::size_t> m_candidateIds;
     mutable std::vector<double> m_candidateMinDistances;
 
+    struct Candidates {
+        std::vector<std::size_t> *candidateIds = nullptr;
+        std::vector<double> *candidateMinDistances = nullptr;
+        std::vector<std::size_t> privateCandidateIds;
+        std::vector<double> privateCandidateMinDistances;
```

Comment:
These should be a unique_ptr of a vector to avoid the allocation. Also, rename it to candidateTdsStorage, candidateMinDistancesStorage.

Reply:
The structure already contains the name candidates, we can remove it from its memebers.

Reply:
Move the declaration before the members.